src/interpreter-stacktrace.c

```diff
@@ -202,6 +202,81 @@ asm(
 #define CALLBACK_ABI  __attribute__((fastcall))
 static_assert(sizeof(interpreter_state) <= MAX_INTERP_STATE_SIZE, "Update assembly code above");
 
+#elif defined(_CPU_AARCH64_)
+
+#define MAX_INTERP_STATE_SIZE 64
+#define STACK_PADDING 16
+
+static_assert(sizeof(interpreter_state) <= MAX_INTERP_STATE_SIZE, "Stack layout invariants violated.");
+static_assert(MAX_INTERP_STATE_SIZE % alignof(interpreter_state) == 0, "Stack layout invariants violated");
+static_assert(((MAX_INTERP_STATE_SIZE + STACK_PADDING) % 16) == 0, "Stack layout invariants violated");
+
+size_t TOTAL_STACK_PADDING = STACK_PADDING;
+
+asm(
+    ASM_ENTRY
+    MANGLE("enter_interpreter_frame") ":\n"
+    ".cfi_startproc\n"
+    "\tstr x30, [sp, #-(" XSTR(MAX_INTERP_STATE_SIZE) " + " XSTR(STACK_PADDING) ")]!\n"
+    "\t.cfi_def_cfa_offset (" XSTR(MAX_INTERP_STATE_SIZE) " + " XSTR(STACK_PADDING) ")\n"
+    "\t.cfi_offset 30, -(" XSTR(MAX_INTERP_STATE_SIZE) " + " XSTR(STACK_PADDING) ")\n"
+    "\tmov x2, x0\n"
+    // Zero out the src and mi fields
```

Comment:
Presumably we should be zeroing out the mi field on the other architectures as well?

Reply:
Yes, I was going to ask you about it. I zerod the mi field here since the reader checks that first. I assume this is only useful for profiling?

Reply:
Jeff added that field after I wrote the backtracing stuff. He may have assumed it's zero initialized.

Reply:
.... OK...

I've added the zeroing to other archs too and also added some comments to the aarch32/aarch64 version.